### PR TITLE
#507 Add MRT destination numbers

### PR DESCRIPTION
**Pull request by langonginc**

### CODE REVIEW

File: src/components/svgs/nodes/mrt-num-line-badge.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                    width="12"
+                    height="12"
+                    fill={fgColor}
+                    fontSize="9"
+                    dominantBaseline="central"
+                    letterSpacing="-0.2"
+                >
+                    {num}
+                </text>
+                {/* Below is an overlay element that has all event hooks but can not be seen. */}
```
Comment (thekingofcity) [inline src/components/svgs/nodes/mrt-num-line-badge.tsx:48]: If we won't connect this node with lines, we should be able to move `onPointerDown`, `onPointerMove`, `onPointerUp`, `style` to the `<g id={id} .../>`.

Comment (langonginc) [reply src/components/svgs/nodes/mrt-num-line-badge.tsx:48]: image

Is this okay?

Comment (thekingofcity) [reply src/components/svgs/nodes/mrt-num-line-badge.tsx:55]: Sure

Comment (thekingofcity) [reply src/components/svgs/nodes/mrt-num-line-badge.tsx:55]: Oh Ln56 should be removed as well as the purpose of a mask element is to provide a `stn_core_${id}` for `document.elementsFromPoint` and pointer events. The first is not needed anymore and the second is moved to the outer g element.

File: src/components/svgs/nodes/mrt-num-line-badge.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        () => (
+            <g
+                id={id}
+                transform={`translate(${x}, ${y})`}
+                onPointerDown={onPointerDown}
+                onPointerMove={onPointerMove}
+                onPointerUp={onPointerUp}
+                style={{ cursor: 'move' }}
+            >
+                <rect fill={bgColor} x="-6" y="-6" width="12" height="12" rx="6" ry="6" />
```
Comment (thekingofcity) [inline src/components/svgs/nodes/mrt-num-line-badge.tsx:40]: Why just not use a `<circle />` element?

Comment (langonginc) [reply src/components/svgs/nodes/mrt-num-line-badge.tsx:40]: I remembered that there would be different between safari and edge/chrome if using circle when I made chongqingRT num line badge.

Comment (thekingofcity) [reply src/components/svgs/nodes/mrt-num-line-badge.tsx:40]: Hummmm, is it possible to upload some screenshots? Safari is a hard browser to set the environment up :(

Comment (langonginc) [reply src/components/svgs/nodes/mrt-num-line-badge.tsx]: I changed it, and no error thrown this time.

**Comment by thekingofcity** [comment]
Just wondering if this is a line badge instead of something like a service badge as it's wired to have one more e.g. 1,2,3 for the East-West line.

**Comment by langonginc** [comment]
> Just wondering if this is a line badge instead of something like a service badge as it's wired to have one more e.g. 1,2,3 for the East-West line.

I'm not sure about Singapore MRT. Maybe ask _Linchen1965_ about it?

**Comment by langonginc** [comment]
image

Hmm, maybe `EWL`, `NSL`, ... is the line badge? But they did not appear on the route map.

**Comment by linchen1965** [comment]
> Just wondering if this is a line badge instead of something like a service badge as it's wired to have one more e.g. 1,2,3 for the East-West line.

They are Destination Numbers to identify the train destinations. All terminal stations on the MRT lines have a destination number assigned to them.

**Comment by thekingofcity** [comment]
> > Just wondering if this is a line badge instead of something like a service badge as it's wired to have one more e.g. 1,2,3 for the East-West line.
> 
> They are Destination Numbers to identify the train destinations. All terminal stations on the MRT lines have a destination number assigned to them.

Thanks for pointing out. 

@langonginc let's rename this node to `MRTDestinationNumbers` and `mrt-dest-num` :)

**Review by thekingofcity — APPROVED** [review]
LGTM

**Pull request merged**